lib/Test2/Harness/Util/IPC.pm

```diff
@@ -105,7 +105,8 @@ sub _run_cmd_fork {
 
     swap_io(\*STDERR, $stderr, $die) if $stderr;
     swap_io(\*STDOUT, $stdout, $die) if $stdout;
-    $stdin ? swap_io(\*STDIN,  $stdin,  $die) : close(STDIN);
```

Comment:
I close STDIN for 2 reasons:
1) If I do not then the test may get keystrokes intended for something else. OR a test may sit there forever waiting for input and nobody types anything, best to close STDIN to make a `<STDIN>` fail. Though it appears that does not work?
2) Not closing STDIN in the test sometimes results in a borked terminal post-tests. I have not been able to figure out when/why it happens, but closing STDIN for test processes made it stop.

So, instead of leaving it open, lets figure out a better solution. Maybe always redirect STDIN to read from an empty file?

Reply:
> Maybe always redirect STDIN to read from an empty file?

That's precisely what I'm doing in L109. `open STDIN, "<", "/dev/null"` closes the old STDIN, then reopens `/dev/null` in its place (which will always return EOF on read).

Reply:
IIRC on some systems, if you leave any of STDIN, STDOUT, or STDERR closed, opening new files may reuse those descriptors.  Attempts to read from STDIN could try to read from a random file opened in the future.